Update homebox to v0.10.1

smolgrrr opened the pull request:
Moved over to version-tagged image as per comment on last update
` TODO: update with version-tagged image when available (https://github.com/hay-kot/homebox/pull/372)`

Tested on:

- [x] x86_64 -> UmbrelOS on proxmox Debian instance
- [x] Arm64 -> Pi 4 8GB

smolgrrr commented:
Just realising I also need to add release notes. Let me know what you think of this, and I'll add

releaseNotes: >-
  This release is a minor release updating dependencies and moving to a Low-Privileged/Distroless Docker Image
  
  
  Read the full release notes for additional information and detailed changes at https://github.com/hay-kot/homebox/releases/tag/v0.10.1

nmfretz commented:
Awesome, thanks @smolgrrr! 

>Moved over to version-tagged image as per comment on last update

For this, we still want to use the "rootless" image so that the Homebox container doesn't have root permissions. See here for context: https://github.com/getumbrel/umbrel-apps/pull/501#issuecomment-1512422781

When we updated this last time, the rootless images did not have version tags yet. But now it looks like they are available (thank you @hay-kot!).  

This is the image we want to use:
https://github.com/hay-kot/homebox/pkgs/container/homebox/136081427?tag=v0.10.1-rootless

smolgrrr commented:
ah i see, thanks. ill test this out now

smolgrrr commented:
Changed to rootless image. Updated, and was nice to see my user info persist 
Retested on:
* [x]  x86_64 -> UmbrelOS on proxmox Debian instance
* [x]  Arm64 -> Pi 4 8GB

nmfretz commented:
Excellent, thanks @smolgrrr! Going live.
